docs/architecture/gas/parameter_definition.md

```diff
@@ -6,8 +6,61 @@ IMPORTANT: This is not the final list of parameters, it contains the base values
+
+### Principles
+Before adding anything, please review the basic principles for gas parameters.
+- A parameter must correspond to a clearly defined workload.
+- When the workload is scalable by a factor `N` that depends on user input,
+  likely it will require a base parameter and a second parameter that is
+  multiplied by `N`. (Example: `N` = number of bytes when reading a value from
+  storage.)
+- Charge gas before executing the workload.
+- Parameters should be independent form specific implementation choices in
+  nearcore.
+- Ideally, contract developers can easily understand what the cost is simply by
+  reading the name in a gas profile.
+
+### Necessary Code Changes
+Adding the parameter in code involves several steps.
+1. Define the parameter by adding it to the list in `parameters.txt.`
+2. Update the Rust view of parameters by adding a variant to `enum Parameter`
+   in `core/primitives-core/src/parameter.rs`. In the same file, update
+   `enum FeeParameter` if you add an action cost or update `ext_costs()`
+   if you add a cost inside function calls.
+3. Update `RuntimeConfig`, the configuration used to reference parameters in
+   code. Depending on the type of parameter, you will need to update
+   `RuntimeFeesConfig` (for action costs) or `ExtCostsConfig` (for gas costs).
+4. Update the list used for gas profiles. This is defined by `enum Cost` in
+   `core/primitives-core/src/profile.rs`. You need to add a variant to either
+   `enum ActionCosts` or `enum ExtCost`. Please also update `fn index()` that
+   maps each profile entry to a unique position in serialized gas profiles.
+5. The parameter should be available to use in the code section you need it. Now
+   is a good time to ensure `cargo check` and `cargo test --no-run` pass. Most
+   likely you have to update some testing code, such as
+   `ExtCostsConfig::test()`.
+6. To merge your changes into nearcore, you will have to hide your parameter
+   behind a feature flag. Add the feature to the `Cargo.toml` of each crate
+   touched in step 3 and 4 and hide the code behind `#[cfg(feature =
+   "protocol_feature_MY_NEW_FEATURE")]`. Do not hide code in step 2 so that
+   non-nightly builds can still read `parameters.txt`. Also add your feature as
+   a dependency on `nightly` in `core/primitives/Cargo.toml` to make sure it
+   gets included when compiling for nightly. After that, check `cargo check` and
+   `cargo test --no-run` with and without `features=nightly`.
+
+### What Gas Value Should the Parameter Have?
+For a first draft, the exact gas value used in the parameter is not crucial.
+Make sure the right set of parameters exists and try to set a number that roughly
+makes sense. This should be enough to enable discussions on the NEP around
+feasibility and usefulness of the proposed feature. If you are not sure, a good
+rule of thumb is 0.1 Tgas for each disk operation and at least 1 Tgas for each 
+ms of CPU time. Then round it up generously.
```

Comment:
do you want to add a reference to this as well? 
https://near.github.io/nearcore/architecture/gas/gas_profile.html

Reply:
Yeah that makes sense to also link to that. But probably better right after the list of principles?